From cf720af446b723b2a9e881161af96e6775bd5a0f Mon Sep 17 00:00:00 2001
From: "Hsia-Jun(Randy) Li" <randy.li@synaptics.com>
Date: Fri, 2 Feb 2024 14:42:34 +0800
Subject: [PATCH 12/12] v4l2bufferpool: raw copying for video frame

We would keep the same memory geometry for video frame.
JIRA:
CSMSSMART-599
---
 subprojects/gst-plugins-good/sys/v4l2/gstv4l2bufferpool.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/subprojects/gst-plugins-good/sys/v4l2/gstv4l2bufferpool.c b/subprojects/gst-plugins-good/sys/v4l2/gstv4l2bufferpool.c
index 7d7bf9f6c3..33e99e526e 100644
--- a/subprojects/gst-plugins-good/sys/v4l2/gstv4l2bufferpool.c
+++ b/subprojects/gst-plugins-good/sys/v4l2/gstv4l2bufferpool.c
@@ -624,6 +624,13 @@ gst_v4l2_buffer_pool_set_config (GstBufferPool * bpool, GstStructure * config)
 
   /* keep a GstVideoInfo with defaults for the when we need to copy */
   gst_video_info_from_caps (&pool->caps_info, caps);
+  for (guint i = 0; i < GST_VIDEO_INFO_N_PLANES(&obj->info); i++) {
+    GST_VIDEO_INFO_PLANE_OFFSET(&pool->caps_info, i) =
+      GST_VIDEO_INFO_PLANE_OFFSET(&obj->info, i);
+    GST_VIDEO_INFO_PLANE_STRIDE(&pool->caps_info, i) =
+      GST_VIDEO_INFO_PLANE_STRIDE(&obj->info, i);
+  }
+  GST_VIDEO_INFO_SIZE(&pool->caps_info) = GST_VIDEO_INFO_SIZE(&obj->info);
 
 done:
   ret = GST_BUFFER_POOL_CLASS (parent_class)->set_config (bpool, config);
-- 
2.17.1

